libsql-server/src/namespace/configurator/helpers.rs

```diff
@@ -84,9 +84,14 @@ pub(super) async fn make_primary_connection_maker(
 
     let bottomless_replicator = match primary_config.bottomless_replication {
         Some(ref options) => {
-            tracing::debug!("Checkpointing before initializing bottomless");
-            crate::replication::primary::logger::checkpoint_db(&db_path.join("data"))?;
-            tracing::debug!("Checkpointed before initializing bottomless");
+            // TODO: figure out why we really need this the fixme above is not clear enough but
+            // disabling this allows us to prevent checkpointing of the wal file.
+            if std::env::var("LIBSQL_DISABLE_INIT_CHECKPOINTING").is_ok() {
```

Comment:
I'm not sure whether I understand this. Isn't the code now doing the initial checkpoint only when `LIBSQL_DISABLE_INIT_CHECKPOINTING` is set to anything? That would mean the initial checkpoint is disabled by default and only enabled when [...]_DISABLE_[...] env variable is set, no?

Reply:
Yeah, this has been fixed and will also produce a warn message when this flag is set.